ndb/src/google/cloud/ndb/_eventloop.py

```diff
@@ -141,18 +145,18 @@ def queue_call(self, delay, callback, *args, **kwargs):
         event = _Event(when, callback, args, kwargs)
         self.insort_event_right(event)
 
-    def queue_rpc(self, rpc, callback=None, *args, **kwds):
-        """Schedule an RPC with an optional callback.
-
-        The caller must have previously sent the call to the service.
-        The optional callback is called with the remaining arguments.
+    def queue_rpc(self, future, rpc):
+        """Add an RPC call.
+        """
+        rpc_id = uuid.uuid1()
```

Comment:
`uuid1` is sufficient here?

Reply:
Why would it not be? (FWIW, legacy actually just used the rpc as a dict key directly. I didn't check to see if that's possible with the futures handed back from grpc, because I thought that was kind of ick. There's no intuitive reason to expect these objects to be hashable on their own.)